### ISSUE TITLE
`borg recreate` fails with `KeyError: 'sparse'` when using `fixed` chunker

### ISSUE THREAD

**Issue by MRWITEK** [borgbackup/borg#7056]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG

#### Your borg version (borg -V).
`borg 1.2.2`

#### Operating system (distribution) and version.
Up-to-date Arch Linux

#### Filesystems used.
ext4 and tmpfs

#### How much data is handled by borg?
Any amount

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
```
borg recreate --chunker-params=fixed,4096 ./borg
```

## Describe the problem you're observing.
`borg recreate` fails with `KeyError: 'sparse'`

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
```
borg init --encryption none ./borg
echo testing | borg create ./borg::data -
borg --debug recreate --chunker-params=fixed,4096 ./borg
```

#### Include any warning/errors/backtraces from the system logs

```
using builtin fallback logging configuration
33 self tests completed in 0.10 seconds
Verified integrity of /tmp/borg/index.5
TAM-verified manifest
security: read previous location '/tmp/borg'
security: read manifest timestamp '2022-09-26T21:24:52.146884'
security: determined newest manifest timestamp as 2022-09-26T21:24:52.146884
security: repository checks ok, allowing access
Verified integrity of /home/cat/.cache/borg/aafd5e0ed92a7d5fe9df2f810d49ce6713b4e4c0704179cbedeaa97616d9ca61/chunks
security: read previous location '/tmp/borg'
security: read manifest timestamp '2022-09-26T21:24:52.146884'
security: determined newest manifest timestamp as 2022-09-26T21:24:52.146884
security: repository checks ok, allowing access
Rechunking archives to ('fixed', 4096, 0)
Processing data
Rechunking archive from ('buzhash', 19, 23, 21, 4095) to ('fixed', 4096, 0)
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 5159, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 5090, in run
    return set_ec(func(args))
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 181, in wrapper
    return method(self, args, repository=repository, cache=cache_, **kwargs)
  File "/usr/lib/python3.10/site-packages/borg/archiver.py", line 1712, in do_recreate
    if not recreater.recreate(name, args.comment):
  File "/usr/lib/python3.10/site-packages/borg/archive.py", line 2121, in recreate
    target = self.create_target(archive, target_name)
  File "/usr/lib/python3.10/site-packages/borg/archive.py", line 2312, in create_target
    target.chunker = get_chunker(*target.chunker_params, seed=self.key.chunk_seed)
  File "src/borg/chunker.pyx", line 288, in borg.chunker.get_chunker
KeyError: 'sparse'

Platform: Linux deskcat 5.19.10-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 20 Sep 2022 15:17:59 +0000 x86_64
Linux: Unknown Linux
Borg: 1.2.2  Python: CPython 3.10.7 msgpack: 1.0.4 fuse: llfuse 1.4.2 [pyfuse3,llfuse]
PID: 26890  CWD: /tmp
sys.argv: ['/usr/bin/borg', '--debug', 'recreate', '--chunker-params=fixed,4096', './borg']
SSH_ORIGINAL_COMMAND: None
```

**Comment by ThomasWaldmann** [comment]
@MRWITEK can you try the fix from #7057 ?

**Comment by ThomasWaldmann** [comment]
I added a test and reproduced the issue. The fix in #7057 fixes the issue in 1.2-maint branch, so it will be in next release, borg 1.2.3.
